### PR TITLE
Initial conformance tests for graph query

### PR DESCRIPTION
**Pull request by vgapeyev**
Adjustments for running graph conformance tests in partiql-tests (as needed for loading external graphs into evaluation environments) and the tests themselves.

The tests come in two variations: unit tests and conformance tests in `partiql-tests` submodule. 

## Other Information
- Updated Unreleased Section in CHANGELOG: NO
  - Not sure; tests are an internal matter?

- Any backward-incompatible changes? NO
- Any new external dependencies? NO

## License Information

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: partiql-lang/src/test/kotlin/org/partiql/lang/eval/EvaluatingCompilerGraphMatchTests.kt [hunk 1]
```diff
@@ -14,7 +14,292 @@ class EvaluatingCompilerGraphMatchTests : EvaluatorTestBase() {
...
+        )
+    )
+
+    @ParameterizedTest
+    @ArgumentsSource(Directionality::class)
+    fun testDirectionality(qr: Pair<String, String>) {
+        testGraphQueries(sessionDirectionality, qr)
+    }
+
+    class Directionality : ArgumentsProviderBase() {
```
Comment (alancai98) [inline partiql-lang/src/test/kotlin/org/partiql/lang/eval/EvaluatingCompilerGraphMatchTests.kt:53]: Will these tests still remain in `partiql-lang-kotlin` after https://github.com/partiql/partiql-tests/pull/91 is merged into `partiql-tests`? Seems like a duplication of tests.

Or perhaps are we waiting for the Kotlin conformance test runner improvements that we spoke about in https://github.com/partiql/partiql-lang-kotlin/pull/1096 before relying just on `partiql-tests`?

File: partiql-lang/src/test/kotlin/org/partiql/lang/eval/EvaluatingCompilerGraphMatchTests.kt [hunk 2]
```diff
@@ -14,7 +14,292 @@ class EvaluatingCompilerGraphMatchTests : EvaluatorTestBase() {
...
+
+            "(N1D1 MATCH (x))" to "<< {'x':1} >>",
+            "(N1D1 MATCH -[y]-> )" to "<< {'y':1.1} >>",
+            "(N1D1 MATCH (x)-[y]->(z) )" to "<< {'x':1, 'y':1.1, 'z':1} >>",
+            "(N1D1 MATCH (x)-[y]->(x) )" to "<< {'x':1, 'y':1.1} >>",
+            "(N1D1 MATCH (x1)-[y1]->(x2)-[y2]->(x3) )" to "<< {'x1':1, 'y1':1.1, 'x2':1, 'y2':1.1, 'x3':1} >>",
+
+            "(N1U1 MATCH (x))" to "<< {'x':1} >>",
+//  Results are not yet correct: need the deduplication of Section 6.5
+//            "(N1U1 MATCH ~[y]~ )" to "<< {'y':1.1} >>",
```
Comment (alancai98) [inline partiql-lang/src/test/kotlin/org/partiql/lang/eval/EvaluatingCompilerGraphMatchTests.kt:169]: Noticed there are some conformance tests failing when using the `partiql-tests` commit w/ the new graph tests (see [conformance report comment](https://github.com/partiql/partiql-lang-kotlin/pull/1104#issuecomment-1563486956)).

Just checking do those failing tests correspond to the commented out tests in this file? The number of failing matches up (12 test failing = 6 commented out tests * 2 evaluation modes tested)

Comment (vgapeyev) [reply partiql-lang/src/test/kotlin/org/partiql/lang/eval/EvaluatingCompilerGraphMatchTests.kt:169]: Yes, those should be due to the commented-out unit tests -- I decided to include them as conformance tests, as our setup is "resilient" to those failing.  

(But if I am lucky, these failures will not last for too long.)

File: partiql-lang/src/test/kotlin/org/partiql/lang/eval/EvaluatingCompilerGraphMatchTests.kt (hunk 1 above)
Comment (alancai98) [reply partiql-lang/src/test/kotlin/org/partiql/lang/eval/EvaluatingCompilerGraphMatchTests.kt:53]: > For tests that have been previously ported from kotlin to conformance, have originals always been removed?

Tests that were part of `pts` were removed (https://github.com/partiql/partiql-lang-kotlin/pull/910). The other tests haven't yet been removed yet (e.g. porting from `EvaluatorTestSuite.kt`.

---

Agree with the point that the test runner would need improvements in running the tests and modeling passing/failing tests before we go through with deleting the other tests.

**Review by alancai98 — DISMISSED** [review]

**Review by jpschorr — APPROVED** [review]

**Review by alancai98 — APPROVED** [review]

**Pull request merged**